app/client/src/components/shared/WaterbodyInfo.tsx

```diff
@@ -1078,6 +1116,570 @@ function MapPopup({
+    setCountCc(
+      selectedData.length ? selectedData.reduce((a, b) => a + b, 0) : null,
+    );
+    setMinCc(getMinCellConcentration(selectedData));
+    setMaxCc(getMaxCellConcentration(selectedData));
+    const newAverageCc = getAverageCellConcentration(selectedData);
+    setAverageCc(newAverageCc);
+    setStdDevCc(getStdDevCellConcentration(selectedData, newAverageCc));
+  }, [cellConcentration, selectedDate]);
+
+  const [projectionLoaded, setProjectionLoaded] = useState<boolean>(false);
+
+  useEffect(() => {
+    projection.load().then(() => setProjectionLoaded(true));
+  }, []);
+
+  const [imageStatus, setImageStatus] = useState<
+    'idle' | 'pending' | 'failure' | 'success'
+  >('idle');
+
+  // Fetch the satellite image for the selected date
+  // and add it to the CyAN Media Layer
+  useEffect(() => {
+    if (selectedDate === null) return;
+    if (services?.status !== 'success') return;
+    if (!mapView) return;
+    if (!projectionLoaded) return;
+
+    const cyanImageLayer = mapView.map.findLayerById('cyanImages');
+    if (!cyanImageLayer || !isMediaLayer(cyanImageLayer)) return;
+
+    const currentDate = new Date(selectedDate);
+    const imageUrl = `${services.data.cyan.images}/?OBJECTID=${
+      attributes.oid ?? attributes.OBJECTID
+    }&year=${currentDate.getFullYear()}&day=${getDayOfYear(currentDate)}`;
+    const propertiesUrl = `${services.data.cyan.properties}/?OBJECTID=${
+      attributes.oid ?? attributes.OBJECTID
+    }`;
+
+    const abortController = new AbortController();
+    const timeout = 60_000;
+    const imageTimeout = setTimeout(() => abortController.abort(), timeout);
+
+    cyanImageLayer.source.elements.removeAll();
+    setImageStatus('pending');
+    const imagePromise = fetch(imageUrl, { signal: abortController.signal });
+    const propsPromise = fetchCheck(propertiesUrl, abortController.signal);
+    Promise.all([imagePromise, propsPromise])
+      .then(([imageRes, propsRes]) => {
+        if (imageRes.headers.get('Content-Type') !== 'image/png') {
+          setImageStatus('idle');
+          return;
+        }
+
+        imageRes.blob().then((blob) => {
+          const image = new Image();
+          image.src = URL.createObjectURL(blob);
+          image.onload = () => setImageStatus('success');
+          const imageElement = new ImageElement({
+            image,
+            georeference: new ExtentAndRotationGeoreference({
+              extent: new Extent({
+                spatialReference: SpatialReference.WGS84,
+                xmin: propsRes.properties.x_min,
+                xmax: propsRes.properties.x_max,
+                ymin: propsRes.properties.y_min,
+                ymax: propsRes.properties.y_max,
+              }),
+            }),
+          });
+          cyanImageLayer.source.elements.add(imageElement);
+        });
+      })
+      .catch((err) => {
+        setImageStatus('failure');
```

Comment:
Can you display an error message on the screen when the `imageStatus === 'failure'`? Currently, it just silently fails if either the `https://qed.epa.gov/waterbody/image` or `https://qed.epa.gov/waterbody/properties` services are down.

Reply:
Good call, I'll add that.